Source/UserManagement/Policies/GreetingGenerators/GreetingGeneratorEventProcessor.cs

```diff
@@ -1,46 +1,45 @@
 using doLittle.Domain;
 using doLittle.Events.Processing;
-using Events;
 using Read.DataCollectors;
+using Domain.MessageGenerator;
+using Events.DataCollector;
 
 namespace Policies.GreetingGenerators
 {
     using Read.GreetingGenerators;
 
     public class GreetingGeneratorEventProcessor : ICanProcessEvents
     {
-
         readonly IGreetingHistories _greetingHistories;
         readonly IDataCollectors _dataCollectors;
-        readonly IAggregateRootRepositoryFor<Domain.MessageGenerators.MessageGenerator> _messageGeneratorsAggregateRootRepository;
+        readonly IAggregateRootRepositoryFor<MessageGenerator> _messageGeneratorsAggregateRootRepository;
 
         public GreetingGeneratorEventProcessor(
             IGreetingHistories greetingHistories,
             IDataCollectors dataCollectors,
-            IAggregateRootRepositoryFor<Domain.MessageGenerators.MessageGenerator> messageGeneratorsAggregateRootRepository)
+            IAggregateRootRepositoryFor<MessageGenerator> messageGeneratorsAggregateRootRepository)
         {
             _greetingHistories = greetingHistories;
             _dataCollectors = dataCollectors;
             _messageGeneratorsAggregateRootRepository = messageGeneratorsAggregateRootRepository;
         }
-
-        public void Process(PhoneNumberAddedToDataCollector @event)
+        public async void Process(PhoneNumberAddedToDataCollector @event)
         {
-            var dataCollector = _dataCollectors.GetById(@event.DataCollectorId);
+            var dataCollector = await _dataCollectors.GetByIdAsync(@event.DataCollectorId);
 
             // Todo Get the correct welcome message based on the dataCollector.PreferredLanguage
             var welcomeMessage = "Welcome!";
 
-            var smsGenerator = _greetingHistories.GetByPhoneNumber(@event.PhoneNumber);
+            var smsGenerator =await  _greetingHistories.GetByPhoneNumberAsync(@event.PhoneNumber);
             if (smsGenerator != null)
             {
                 return;
             }
 
-            var smsGeneratorAggregateRootRepository = _messageGeneratorsAggregateRootRepository.Get(@event.Id);
-            smsGeneratorAggregateRootRepository.GenerateMessage(new Domain.MessageGenerators.GenerateMessage()
+            var smsGeneratorAggregateRootRepository = _messageGeneratorsAggregateRootRepository.Get(@event.DataCollectorId);
```

Comment:
I think this is due to a limitation in the infrastructure, but I'd rather that the policy could be an Event Source and generate the event.  MessageGenerator as an AggregateRoot just feels wrong.